### PR TITLE
libc: Add the empty nl_langinfo implementation

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/langinfo.h.html
Note: The implementation match what we have done for locale function.

## Impact
New API

### CODE REVIEW

File: include/langinfo.h [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+#include <locale.h>
+
+#define NL_ITEM(cat, index) (((cat) << 16) | (index))
+#define NL_LOCALE_NAME(cat) NL_ITEM(cat，0xffff)
+
+/****************************************************************************
+ * Public Type Definitions
+ ****************************************************************************/
+
+enum
```
Comment (gustavonihei) [inline include/langinfo.h:38]: The coding standard forbids the declaration of un-names enumerations.

File: include/langinfo.h [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+ ****************************************************************************/
+
+#include <nl_types.h>
+#include <locale.h>
+
+#define NL_ITEM(cat, index) (((cat) << 16) | (index))
+#define NL_LOCALE_NAME(cat) NL_ITEM(cat，0xffff)
+
+/****************************************************************************
+ * Public Type Definitions
```
Comment (gustavonihei) [inline include/langinfo.h:35]:
```suggestion
 * Public Types
```

File: include/langinfo.h [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+#ifndef __INCLUDE_LANGINFO_H
+#define __INCLUDE_LANGINFO_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nl_types.h>
+#include <locale.h>
+
```
Comment (gustavonihei) [inline include/langinfo.h:30]:
```suggestion
/****************************************************************************
 * Pre-processor Definitions
 ****************************************************************************/
 
```

File: include/langinfo.h (hunk 1 above)
Comment (xiaoxiang781216) [reply include/langinfo.h:38]: Done.

File: include/langinfo.h (hunk 2 above)
Comment (xiaoxiang781216) [reply include/langinfo.h:35]: Done.

File: include/langinfo.h (hunk 3 above)
Comment (xiaoxiang781216) [reply include/langinfo.h:30]: Done.

File: include/langinfo.h [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+#ifdef __cplusplus
+#define EXTERN extern "C"
+extern "C"
+{
+#else
+#define EXTERN extern
+#endif
+
+FAR char *nl_langinfo(nl_item item);
+FAR char *nl_langinfo_l(nl_item item, locale_t l);
```
Comment (gustavonihei) [inline include/langinfo.h:119]: `nl_langinfo_l` Is already implemented as a macro, this prototype declaration should not be required.

Comment (xiaoxiang781216) [reply include/langinfo.h]: Done.

**Review by gustavonihei — APPROVED** [review]
LGTM

**Pull request merged**